### PR TITLE
Result of the running inspections and fixing meaningful warnings

### PR DESCRIPTION
**Pull request by ermolenkodev**
I have used this inspection profile
<details>

```
<component name="InspectionProjectProfileManager">
  <profile version="1.0">
    <option name="myName" value="Project Default" />
    <inspection_tool class="Destructure" enabled="true" level="INFO" enabled_by_default="true" />
    <inspection_tool class="DirectUseOfResultType" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="EmptyDirectory" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="FromClosedRangeMigration" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="IncompleteDestructuring" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="InconsistentLineSeparators" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="IncorrectFormatting" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="KDocMissingDocumentation" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="KotlinLoggerInitializedWithForeignClass" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="LongLine" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="NonExhaustiveWhenStatementMigration" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="ProblematicWhitespace" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="ProhibitJvmOverloadsOnConstructorsOfAnnotationClassesMigration" enabled="true" level="ERROR" enabled_by_default="true" />
    <inspection_tool class="ProhibitRepeatedUseSiteTargetAnnotationsMigration" enabled="true" level="ERROR" enabled_by_default="true" />
    <inspection_tool class="ProhibitTypeParametersForLocalVariablesMigration" enabled="true" level="ERROR" enabled_by_default="true" />
    <inspection_tool class="ProhibitUseSiteTargetAnnotationsOnSuperTypesMigration" enabled="true" level="ERROR" enabled_by_default="true" />
    <inspection_tool class="PublicApiImplicitType" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="RedundantInnerClassModifier" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="RedundantLabelMigration" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="RedundantNotNullExtensionReceiverOfInline" enabled="true" level="INFORMATION" enabled_by_default="true" />
    <inspection_tool class="ReplaceCollectionCountWithSize" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="ReplaceNotNullAssertionWithElvisReturn" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="RestrictReturnStatementTargetMigration" enabled="true" level="ERROR" enabled_by_default="true" />
    <inspection_tool class="TodoComment" enabled="true" level="WARNING" enabled_by_default="true" />
    <inspection_tool class="UnusedMainParameter" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="WarningOnMainUnusedParameterMigration" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
    <inspection_tool class="WhenWithOnlyElse" enabled="true" level="WEAK WARNING" enabled_by_default="true" />
  </profile>
</component>
```

</details>

Many issues were found, but not all of them looked useful, so I fixed only part of the warnings.
Fixes mostly fall into one of these categories:
* Fix typos reported by IDEA inspection
* Naming convention fixes
* Add missed KDocs
* Remove unused code
* Add KDocs for public declarations
* Fix formatting warnings

### CODE REVIEW

File: examples/src/main/kotlin/examples/onnx/cv/efficicentnet/EfficientNet4Lite.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ import java.io.File
  * This examples demonstrates the inference concept on EfficientNet4Lite model:
  * - Model configuration, model weights and labels are obtained from [ONNXModelHub].
  * - Model predicts on a few images located in resources.
- * - Special preprocessing (used in EfficientNet4Lite during training on ImageNet dataset) is applied to images before prediction.
+ * - Special preprocessing (used in EfficientNet4Lite during training on ImageNet dataset) is applied to image before prediction.
```
Comment (juliabeliaeva) [inline examples/src/main/kotlin/examples/onnx/cv/efficicentnet/EfficientNet4Lite.kt:26]: Why "image" is better than "images"? "each image" also seems to fit here. (same for other places)

**Review by juliabeliaeva — APPROVED** [review]

**Comment by ermolenkodev** [comment]
Could you please run a CI? Just to be sure there are no unexpected side-effects

File: impl/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/impl/preprocessing/OperationEx.kt [hunk 2]
```diff
@@ -13,15 +13,17 @@ import org.jetbrains.kotlinx.dl.api.preprocessing.PreprocessingPipeline
  * Convenience functions for executing custom logic after applying [Operation].
  * Could be useful for debugging purposes.
  */
-public fun <I, O> Operation<I,O>.onResult(block: (O) -> Unit): Operation<I, O> {
+public fun <I, O> Operation<I, O>.onResult(block: (O) -> Unit): Operation<I, O> {
     return PreprocessingPipeline(this, object : Operation<O, O> {
+        @Suppress("ReturnInsideFinallyBlock")
```
[the remark this answers is not in the document]
Comment (ermolenkodev) [reply impl/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/impl/preprocessing/OperationEx.kt]: @juliabeliaeva

[the remark this answers is not in the document]
Comment (juliabeliaeva) [reply impl/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/impl/preprocessing/OperationEx.kt]: Ok, let's not suppress the exception here.

**Pull request merged**